src/test/suite/lib/ggshield-api.test.ts

```diff
@@ -0,0 +1,70 @@
+import * as simple from 'simple-mock';
+import * as assert from 'assert';
+import { GGShieldConfiguration } from '../../../lib/ggshield-configuration';
+import * as runGGShield from '../../../lib/run-ggshield';
+import { ggshieldScanFile } from '../../../lib/ggshield-api';
+import { window } from 'vscode';
+import { GGShieldScanResults } from "../../../lib/api-types";
+
+suite("ggshieldScanFile", () => {
+    let runGGShieldMock: simple.Stub<Function>;
+    let errorMessageMock: simple.Stub<Function>;
+
+    setup(() => {
+        runGGShieldMock = simple.mock(runGGShield, "runGGShieldCommand");
+        errorMessageMock = simple.mock(window, "showErrorMessage");
+    });
+
+    teardown(() => {
+        simple.restore();
+    });
+
+    test("returns an empty result when ggshield returns no secrets found", () => {
+        runGGShieldMock.returnWith({
+            status: 0,
+            stdout: '{}',
+            stderr: ''
+        });
+
+        const result = ggshieldScanFile('test', new GGShieldConfiguration(
+            '',
+            '',
+            '',
+        ));
+
+        assert.deepStrictEqual(result, {});
+    });
+
+    test("displays an error message when ggshield returns an error", () => {
+        runGGShieldMock.returnWith({
+            status: 2,
+            stdout: '',
+            stderr: 'Invalid API key.'
+        });
+
+        ggshieldScanFile('test', new GGShieldConfiguration(
```

Comment:
I think you can create a mock configuration for all your calls I think (or even replace by something like `{} as GGShieldConfiguration`, you mock ggshield call anyway)